spec/services/push_notifications/send_spec.rb

```diff
@@ -59,31 +60,33 @@
   end
 
   context "with devices for multiple users" do
-    let(:consumer_app) { create(:consumer_app) }
+    let(:consumer_app_ios) { create(:consumer_app, platform: Device::IOS) }
+    let(:consumer_app_android) { create(:consumer_app, platform: Device::ANDROID) }
 
     before do
-      create(:device, user: user, consumer_app: consumer_app)
-      create(:device, user: user2, consumer_app: consumer_app)
+      create(:device, user: user, consumer_app: consumer_app_ios)
+      create(:device, user: user2, consumer_app: consumer_app_ios)
+      create(:device, user: user2, consumer_app: consumer_app_android, platform: Device::ANDROID)
     end
 
     it "creates a notification and enqueues it" do
-      mocked_objects = mock_rpush(consumer_app)
+      mocked_objects = mock_rpush(consumer_app_ios)
 
-      described_class.call(**many_targets_params)
+      described_class.call(**params)
 
-      expect(mocked_objects[:rpush_notification]).to have_received(:save!).exactly(2).times
+      expect(mocked_objects[:rpush_notification]).to have_received(:save!).exactly(1).times
 
       sidekiq_assert_enqueued_jobs(1, only: PushNotifications::DeliverWorker)
     end
 
     it "creates a single notification for each of the user's devices when they have multiple" do
-      create(:device, user: user)
-
-      mocked_objects = mock_rpush(consumer_app)
+      ios_mocked_objects = mock_rpush(consumer_app_ios)
+      android_mocked_objects = mock_rpush(consumer_app_android)
 
       described_class.call(**many_targets_params)
 
-      expect(mocked_objects[:rpush_notification]).to have_received(:save!).exactly(3).times
+      expect(ios_mocked_objects[:rpush_notification]).to have_received(:save!).exactly(2).times
+      expect(android_mocked_objects[:rpush_notification]).to have_received(:save!).exactly(1).times
```

Comment:
Now we're testing both platforms to avoid regressions here. They use separate Rpush model instances (Redis backed models) so they need to be mocked and checked separately.